sklearn/utils/testing.py

```diff
@@ -625,8 +625,9 @@ def run_test(*args, **kwargs):
             import matplotlib
             matplotlib.use('Agg', warn=False)
             # this fails if no $DISPLAY specified
-            matplotlib.pylab.figure()
-        except:
+            import matplotlib.pyplot as plt
+            plt.figure()
+        except ImportError:
```

Comment:
`plt.figure()` can throw any kind of exceptions depending on the matplotlib backend it seems. The following snippet raises TclError with TkAgg and RuntimeError with Qt4Agg.

```python
import os

import matplotlib

display = os.environ.get('DISPLAY')

if display is not None:
    del os.environ['DISPLAY']

# backend = 'TkAgg'
backend = 'Qt4Agg'
matplotlib.use(backend)

import matplotlib.pyplot as plt

# This line does not succeed in changing the backend because
# matplotlib.pyplot has already been imported
matplotlib.use('Agg')
plt.figure()
```

It looks like the simplest is to catch Exception possibly in a separate except clause and raise SkipTest('Unable to create a matplotlib figure').